### PR TITLE
Null Check PickUpAnimation And Other Error Fixes

### PR DESCRIPTION
**Pull request by ThatDan123**
image

Was the last error before being disconnected every time, hopefully this fixes the disconnects.

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
Is adding a null check really the solution here?

Also how can interactor ever be null?

**Comment by ThatDan123** [comment]
It errors on join, in the RPC. Probably cus the client hasnt loaded the object before the message is sent. Not sure why its force closing the connection though.

**Comment by ThatDan123** [comment]
The disconnect is cus is theres an error downstream mirror will force disconnect:

image

I managed to do a check earlier for netmessages to stop those disconnects but since this is an RPC mirror directly handles it

**Comment by MaxIsJoe** [comment]
Ah, alright then.

**Comment by ThatDan123** [comment]
Good to be merged

File: UnityProject/Assets/Scripts/Core/Lighting/EmergencyLightAnimator.cs [hunk 1]
```diff
@@ -47,10 +47,15 @@ protected virtual void UpdateMe()
...
 			if (lightSource == null || lightSource.mLightRendererObject == null ||
-				lightSource.mLightRendererObject.transform == null)
+			    lightSource.mLightRendererObject.transform == null)
 			{
-				Debug.LogError($"{gameObject.name} had something null");
 				StopAnimation();
+
+				if (this != null && gameObject != null)
+				{
+					Debug.LogError($"{gameObject.name} had something null");
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Lighting/EmergencyLightAnimator.cs]:
```suggestion
					Logger.LogError($"{gameObject.name} had something null");
```

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Lighting/EmergencyLightAnimator.cs]: the log message seems a little ambiguous

**Pull request merged**